test/drlg_l1_test.cpp

```diff
@@ -19,3 +26,134 @@ TEST(Drlg_l1, DRLG_Init_Globals)
 	DRLG_Init_Globals();
 	EXPECT_EQ(dLight[0][0], 0);
 }
+
+void TestCreateL5Dungeon(bool hellfire, int level, int seed, lvl_entry entry)
+{
+	pMegaTiles = std::make_unique<MegaTile[]>(1648);
+
+	MyPlayer->pOriginalCathedral = !hellfire;
+
+	currlevel = level;
+	leveltype = DTYPE_CATHEDRAL;
+
+	CreateL5Dungeon(seed, entry);
+
+	std::string path = paths::BasePath();
+
+	paths::SetPrefPath(path);
+	std::string dunPath;
+	if (hellfire)
+		dunPath = fmt::format("../test/fixtures/hellfire/{}-{}.dun", level, seed);
+	else
+		dunPath = fmt::format("../test/fixtures/diablo/{}-{}.dun", level, seed);
+	auto dunData = LoadFileInMem<uint16_t>(dunPath.c_str());
+	EXPECT_EQ(DMAXX, dunData[0]);
+	EXPECT_EQ(DMAXY, dunData[1]);
+
+	const uint16_t *tileLayer = &dunData[2];
+
+	for (int y = 0; y < DMAXY; y++) {
+		for (int x = 0; x < DMAXX; x++) {
+			auto tileId = static_cast<uint8_t>(SDL_SwapLE16(*tileLayer));
+			tileLayer++;
+			EXPECT_EQ(dungeon[x][y], tileId);
+		}
+	}
+
+	const uint16_t *transparentLayer = &dunData[2 + DMAXX * DMAXY * 13];
+
+	for (int y = 16; y < 16 + DMAXY * 2; y++) {
+		for (int x = 16; x < 16 + DMAXX * 2; x++) {
+			auto sectorId = static_cast<uint8_t>(SDL_SwapLE16(*transparentLayer));
+			transparentLayer++;
+			EXPECT_EQ(dTransVal[x][y], sectorId);
+		}
+	}
+}
+
+TEST(Drlg_l1, CreateL5Dungeon_diablo_1_743271966)
+{
+	TestCreateL5Dungeon(false, 1, 743271966, ENTRY_MAIN);
+	EXPECT_EQ(ViewPosition.x, 51);
+	EXPECT_EQ(ViewPosition.y, 82);
+	TestCreateL5Dungeon(false, 1, 743271966, ENTRY_PREV);
+	EXPECT_EQ(ViewPosition.x, 81);
+	EXPECT_EQ(ViewPosition.y, 47);
+}
+
+TEST(Drlg_l1, CreateL5Dungeon_diablo_2_1383137027)

```

Comment:
Having a version of these quest level tests with the other state (diablo unavailable, hellfire available) would be great (in a later pr :D)

Reply:
Yeah, was a bit random with what I exported since there was still a lot to figure out regarding what and how to test things. So yeah, very much meant to have several follow up PRs once we have the basics down :)